blog/posts/2019-10-22-shepherd-popper-to-tether.md

```diff
@@ -0,0 +1,56 @@
+---
+authorId: rwwagner90
+categories: 
+  - app tour
+  - shepherd.js
+  - site tour
+date: '2019-10-22'
+nextSlug: exporting-typescript-imports
+nextTitle: 'Exporting Imported TypeScript Types'
+previousSlug: ember-data-passing-query-params-to-save
+previousTitle: Ember Data | Passing query params to .save()
+slug: shepherd-popper-to-tether
+title: 'There and Back Again - Popper to Tether'
+---
+
+[Shepherd.js](https://shepherdjs.dev/) has gone through many iterations, 
+since we started refactoring it for v2. We initially switched from 
+[Tether](https://github.com/shipshapecode/tether) to [Popper](https://popper.js.org/), on the 
+advice of the Tether maintainers who said they did not have time to maintain Tether
+and that Popper was a better option for now. We then doubled down and switched
+to [Tippy.js](https://atomiks.github.io/tippyjs/) which is built on top of Popper, and 
+provides animations and extra features. Ultimately, we ended up with a really bloated bundle, 
+with lots of dependencies, and we decided to try to trim the fat.
+
+We removed `lodash`, `object-assign-deep`, and other various dependencies, in favor
+of shipping small util methods we wrote internally, and got down to Tippy being
+our only dependency. We then decided we did not need the extra functionality that
+Tippy offered, and we refactored back to using just Popper, which removed a pretty
+sizable chunk from our bundle.
+
+We wanted to keep saving more though, since Popper has become a pretty in depth
+framework, and might be more than we needed. This led us to try to create our
+own positioning functionality, and explore other libs, and we ultimately had an
+idea, why not try Tether again? For those who don't know, Shepherd was initially
```

Comment:
```suggestion
idea, why not try Tether again? Since Shepherd was initially
```

We already mentioned it was Tether first in the initial paragraph.